cloudevents-sdk-warp/src/server_response.rs

```diff
@@ -0,0 +1,95 @@
+use super::headers;
+
+use cloudevents::event::SpecVersion;
+use cloudevents::message::{
+    BinaryDeserializer, BinarySerializer, Error, MessageAttributeValue, Result,
+    StructuredSerializer,
+};
+use cloudevents::Event;
+
+use warp::http::HeaderValue;
+use warp::hyper::Body;
+use warp::reply::Response;
+
+use http::header::HeaderName;
+use http::response::Builder;
+
+use std::str::FromStr;
+
+pub struct ResponseSerializer {
+    builder: Builder,
+}
+
+impl ResponseSerializer {
+    // #[allow(dead_code)]
+    // pub fn from(builder: Builder) -> ResponseSerializer {
+    //     ResponseSerializer { builder }
+    // }
+
+    fn new() -> Self {
+        ResponseSerializer {
+            builder: http::Response::builder(),
+        }
+    }
+}
+
+impl BinarySerializer<Response> for ResponseSerializer {
+    fn set_spec_version(mut self, spec_version: SpecVersion) -> Result<Self> {
+        self.builder = self.builder.header(
+            headers::SPEC_VERSION_HEADER.clone(),
+            str_to_header_value!(spec_version.as_str())?,
+        );
+        Ok(self)
+    }
+
+    fn set_attribute(mut self, name: &str, value: MessageAttributeValue) -> Result<Self> {
+        self.builder = self.builder.header(
+            headers::ATTRIBUTES_TO_HEADERS.get(name).unwrap().clone(),
+            str_to_header_value!(value.to_string().as_str())?,
```

Comment:
More than this (which eventually leads to another copy), can you do `HeaderValue::try_from` passing directly the string? (same below)